### DISCUSSION TITLE
Support Table `CHECK` Constraints

### DISCUSSION THREAD

**Discussion by forrest-akin** [SeaQL/sea-query#135] [Q&A] [answered]
Currently, there's no way to specify table `CHECK` constraints in [`TableCreateStatement`](https://docs.rs/sea-query/0.12.8/sea_query/table/struct.TableCreateStatement.html#)s. Adding an `.extra()` method similar to [`ColumnDef`](https://docs.rs/sea-query/0.12.8/sea_query/table/struct.ColumnDef.html#method.extra) would probably be the easiest way to add support for this.

**Comment by billy1624** [comment]
You can specify unique, primary key and reference constraints with `index`, `primary_key` and `foreign_key` methods in `TableCreateStatement`.

**Comment by billy1624** [comment]
However, yesterday I discovered the generated unique constraint SQL statement for Postgres is incorrect. There will be an excess "KEY" keyword in the statement. #85 

```rust
    Table::create()
        .table(Alias::new("film"))
        .col(ColumnDef::new(Alias::new("title")).char_len(255).not_null())
        .col(ColumnDef::new(Alias::new("description")).text())
        .primary_key(Index::create().primary().col(Alias::new("film_id")))
        .index(Index::create().unique().col(Alias::new("title")))
        .to_owned()
```

**Comment by billy1624** [comment]
You can checkout this example, https://github.com/SeaQL/sea-schema/blob/e284b5b72c098f3afab5600f848245fa28127927/tests/live/postgres/src/main.rs#L75-L162.

**Comment by forrest-akin** [comment]
Oops, I meant to specify table [`CHECK` constraints](https://www.postgresql.org/docs/current/ddl-constraints.html#DDL-CONSTRAINTS-CHECK-CONSTRAINTS) is what's missing. I did run into the unique key issue, but I think the rest is fine as you mentioned.

**Comment by billy1624** [comment, marked as answer]
Hey @forrest-akin, sorry for the delay. I've just open a tracking issue for this feature request :D
- https://github.com/SeaQL/sea-query/issues/397

**Comment by diptanu** [comment]
Hi is this feature not released? I see the issue linked above and the PRs are done but don't have it on the latest crate.
